apps/query-ui/queryui/util.py

```diff
@@ -261,11 +268,31 @@ class QueryTrace:
 
     def __init__(self, trace_dir: str):
         self.trace_dir = trace_dir
-        self.node_traces = [QueryNodeTrace(trace_dir, node_id) for node_id in sorted(os.listdir(self.trace_dir))]
+        self.node_traces = []
+        self.query_plan = self._get_query_plan(self.trace_dir)
+        for dir in sorted(os.listdir(self.trace_dir)):
+            if "metadata" not in dir:
+                self.node_traces += [QueryNodeTrace(trace_dir, dir)]
+
+    def _get_query_plan(self, trace_dir: str):
+        metadata_dir = os.path.join(trace_dir, "metadata")
+        if os.path.isdir(metadata_dir):
```

Comment:
Nit: I think you can just do this in one step, i.e.,

`if os.path.isfile(os.path.join(trace_dir, "metadata", "query_plan.json")):`